ghcide/src/Development/IDE/GHC/Compat/Core.hs

```diff
@@ -931,49 +920,22 @@ pattern L l a <- GHC.L (getLoc -> l) a
 {-# COMPLETE L #-}
 #endif
 
-#elif MIN_VERSION_ghc(8,8,0)
+#else
 type HasSrcSpan = SrcLoc.HasSrcSpan
 getLoc :: SrcLoc.HasSrcSpan a => a -> SrcLoc.SrcSpan
 getLoc = SrcLoc.getLoc
-
-#else
-
-class HasSrcSpan a where
-    getLoc :: a -> SrcSpan
-instance HasSrcSpan Name where
-    getLoc = nameSrcSpan
-instance HasSrcSpan (SrcLoc.GenLocated SrcSpan a) where
-    getLoc = SrcLoc.getLoc
-
 #endif
 
 getRealSrcSpan :: SrcLoc.RealLocated a -> SrcLoc.RealSrcSpan
-#if !MIN_VERSION_ghc(8,8,0)
-getRealSrcSpan = SrcLoc.getLoc
-#else
 getRealSrcSpan = SrcLoc.getRealSrcSpan
-#endif
-
 
 -- | Add the @-boot@ suffix to all output file paths associated with the
 -- module, not including the input file itself
 addBootSuffixLocnOut :: GHC.ModLocation -> GHC.ModLocation
-#if !MIN_VERSION_ghc(8,8,0)
-addBootSuffixLocnOut locn
-  = locn { Module.ml_hi_file  = Module.addBootSuffix (Module.ml_hi_file locn)
-         , Module.ml_obj_file = Module.addBootSuffix (Module.ml_obj_file locn)
-         }
-#else
 addBootSuffixLocnOut = Module.addBootSuffixLocnOut
-#endif
-
 
 dataConExTyCoVars :: DataCon -> [TyCoVar]
-#if __GLASGOW_HASKELL__ >= 808
 dataConExTyCoVars = DataCon.dataConExTyCoVars
```

Comment:
I think some things like this could probably be inlined now?

Reply:
Thanks for these hints :+1: